model/src/main/java/org/openremote/model/webhook/Webhook.java

```diff
@@ -0,0 +1,111 @@
+package org.openremote.model.webhook;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Webhook {
+
+    public enum Method {
+        GET, POST, PUT, DELETE
+    }
+
+    public enum AuthMethod {
+        NONE, HTTP_BASIC, API_KEY, OAUTH2
```

Comment:
HTTP_BASIC and API_KEY can be handled as headers and/or query parameters.
You should just support an `OAuthGrant` being defined for OAuth authentication which needs special handling